test/dynamo/test_higher_order_ops.py

```diff
@@ -621,6 +621,154 @@ def fn(x):
 
         self.assertTrue(activations.keys() == forward_handles.keys())
 
+    def _grad_compile_check(self, fn, *inputs):
+        actual = fn(*inputs)
+        expected = torch.compile(fn)(*inputs)
```

Comment:
testing advice:
- use torch.compile(fn, backend='aot_eager'), we are trying to test that dynamo works (and we don't really care about what inductor does to our code).
- use fullgraph=True. I like this a bit better than checking unique_graphs == 1, but more concretely, this is closer in line with what the rest of this file does
- We should do what the wrap() tests do and check that the graph actually looks like what we expect it to. E.g. there is a GradFunc node somewhere that gets invoked